asf_heat_pump_suitability/getters/get_target.py

```diff
@@ -78,50 +90,128 @@ def get_df_target_property_type_uncensored() -> pl.DataFrame:
         )
         .rename(
             {
-                "Detached": "Detached whole house or bungalow",
-                "Semi-detached": "Semi-detached whole house or bungalow",
-                "Terraced": "Terraced (including end-terrace) whole house or bungalow",
+                "Terraced": "Terraced (including end-terrace)",
+                "A caravan or other mobile or temporary structure": "Caravan or other mobile or temporary structure",
             }
         )
     )
 
     return df
 
 
-def get_df_target_property_type(fill_censored: int = 1) -> pl.DataFrame:
+def load_transform_df_target_property_type_scotland() -> pl.DataFrame:
     """
-    Get dataframe of property type counts for all LSOAs in England and Wales, and fill censored values (counts below 10)
-    with given constant. Source: census data 2021.
-
-    Args:
-        fill_censored (int): value to fill censored values with, [0-10]. Default 0.
+    Load and transform dataframe of property type counts for data zones in Scotland from census data.
 
     Returns:
-        pl.Dataframe: counts of property type for all LSOAs in England and Wales
+        pl.Dataframe: counts of property type for all data zones in Scotland
     """
-    content = base_getters.get_content_from_s3_path(
-        config["data_source"]["EW_census_housing_characteristics"]
+    df = pl.read_csv(
+        config["data_source"]["S_census_accommodation_type"],
+        skip_rows=10,
+        columns=list(range(0, 11)),
+        infer_schema_length=10000,
     )
-    df = pl.read_excel(content, sheet_name="2c", engine="calamine")
-
-    # Remove empty header rows
     df = (
-        df.rename(df[2].to_dicts().pop())
-        .slice(
-            3,
+        df[1:]
+        .drop_nulls(subset=cs.numeric())
+        .drop(
+            [
+                "Whole house or bungalow: Total",
+                "Flat, maisonette or apartment: Total",
+                "All occupied households",
+            ]
+        )
+    )
+    flats_cols = [col for col in df.columns if "Flat" in col]
+    df = (
+        df.with_columns(
+            pl.sum_horizontal(flats_cols).alias("Flat, maisonette or apartment")
+        )
+        .drop(flats_cols)
+        .rename(
+            {
+                col: col.replace("Whole house or bungalow: ", "")
+                for col in df.select(cs.numeric()).columns
+            }
         )
-        .drop(["Area Name"])
-        .rename({"Area Code": "lsoa"})
+        .rename({"Type of accomodation": "lsoa"})
```

Comment:
Thanks for flagging this because it is weird! I double checked it, and it is correct that we rename this to LSOA.

Basically, the original csv file is poorly formatted with a lot of deadspace. This is why this function does a lot of reformatting of the dataframe itself. The reason why this column ends up with the "Type of accommodation" label, is I think because "Type of accommodation" is the name of the columns index. Because of bad formatting it just ends up being the header of the LSOA column and the actual column header ends up in the first row of data.